runtime/src/main/java/org/corfudb/runtime/view/ManagementView.java

```diff
@@ -221,16 +223,18 @@ private ClusterStatus getClusterHealth(Layout layout, Set<String> peerResponsive
 
         // Using the peer views from the nodes to determine health of the cluster.
         Set<String> peerResponsiveNodes = new HashSet<>(layout.getAllServers());
-        nodeViewMap.values().stream()
-                .map(NodeView::getNetworkMetrics)
-                // Get all peer connectivity maps from all node views.
-                .map(NetworkMetrics::getPeerConnectivityView)
-                .flatMap(stringBooleanMap -> stringBooleanMap.entrySet().stream())
+        // Truncates all nodes which did not respond to the Heartbeat request.
+        peerResponsiveNodes.retainAll(nodeViewMap.keySet());
+        // Filter out all nodes which are not reachable by the cluster nodes.
+        nodeViewMap.forEach((endpoint, nodeView) -> nodeView.getNetworkMetrics()
```

Comment:
Can we write this in more imperative style. It is hard to read such a long chain of methods.